lib/utils/docker.coffee

```diff
@@ -182,32 +209,55 @@ exports.runBuild = (params, options, getBundleInfo, logStreams) ->
 
 				buildFailure: reject
 				buildStream: (stream) ->
+					if options.emulated
+						logging.logInfo(logStreams, 'Running emulated build')
+
 					getBundleInfo(options)
 					.then (info) ->
 						if !info?
 							logging.logWarn logStreams, '''
 								Warning: No architecture/device type or application information provided.
 									Dockerfile/project pre-processing will not be performed.
 							'''
-							tarStream.pipe(stream)
+							return tarStream
 						else
 							[arch, deviceType] = info
 							# Perform type resolution on the project
 							bundle = new resolver.Bundle(tarStream, deviceType, arch)
 							resolver.resolveBundle(bundle, resolver.getDefaultResolvers())
 							.then (resolved) ->
 								logging.logInfo(logStreams, "Building #{resolved.projectType} project")
-								# Send the resolved tar stream to the docker daemon
-								resolved.tarStream.pipe(stream)
+
+								return resolved.tarStream
+					.then (buildStream) ->
+						# if we need emulation
+						if options.emulated and platformNeedsQemu()
+							return transpose.transposeTarStream buildStream,
+								hostQemuPath: qemuPath
+								containerQemuPath: './qemu-execve'
+						else
+							return buildStream
+					.then (buildStream) ->
+						# Send the resolved tar stream to the docker daemon
+						buildStream.pipe(stream)
 					.catch(reject)
 
 					# And print the output
-					throughStream = es.through (data) ->
+					logThroughStream = es.through (data) ->
 						logs += data.toString()
 						this.emit('data', data)
 
-					stream
-					.pipe(throughStream)
+					if options.emulated and platformNeedsQemu()
+						buildThroughStream = transpose.getBuildThroughStream
+							hostQemuPath: qemuPath
+							containerQemuPath: './qemu-execve'
```

Comment:
`qemu-execve` is hardcoded all over the place, should it be moved to a constant?